### ISSUE TITLE
RX 580 fails to produce any good shares with xmr stak 2.4.7 but works fine with 2.4.3

### ISSUE THREAD

**Issue by 12345678911156** [fireice-uk/xmr-stak#1894]
rx 580 fails to produce any good shares with xmr stak 2.4.7 but works fine with 2.4.3
AM invalid result GPU ID 0
Everything is default settings from using the provided setups.

Basic information
ryzen 5 2600
rx 580 8gb
Windows 10 x64
AMD GPU Drivers: 18.5.1

GPU passes memtestCL-1.00-windows without errors.

# Stability issue
- Is the CPU or GPU overclocked? no
- Is the Main memory of the CPU or GPU undervolted? no
under-clocking does not solve the problem

**Comment by leve1ord** [comment]
Same with me. Xmr stak 2.4.7 + rx580 = 50% invalid results GPU ID 0. 
Same card with claymore cryptonight 11.3 works like a charm. Funny thing is that rx480 works absolutely stable with same xmr-stak and with same drivers.

**Comment by psychocrypt** [comment]
this is a driver issue. If it works with older gpus the reason is that amd
used internally different compiler for each gpu generation. Please
downgrade to 18.3 in windows.

**Comment by leve1ord** [comment]
Nope, this isn't a driver issue. Tried lots of them from that blockchain to latest adrenalin. Only moving to xmr-stak 2.5.0 fixes problem.
